<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [Mono-bugs] [Bug 549926] Problem with exception handling with async (Web) Service calls
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:mono-bugs%40lists.ximian.com?Subject=%5BMono-bugs%5D%20%5BBug%20549926%5D%20Problem%20with%20exception%20handling%20with%20async%0A%20%28Web%29%20Service%20calls&In-Reply-To=bug-549926-28286%40http.bugzilla.novell.com/">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="094530.html">
   <LINK REL="Next"  HREF="094532.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Mono-bugs] [Bug 549926] Problem with exception handling with async (Web) Service calls</H1>
    <B>bugzilla_noreply at novell.com</B> 
    <A HREF="mailto:mono-bugs%40lists.ximian.com?Subject=%5BMono-bugs%5D%20%5BBug%20549926%5D%20Problem%20with%20exception%20handling%20with%20async%0A%20%28Web%29%20Service%20calls&In-Reply-To=bug-549926-28286%40http.bugzilla.novell.com/"
       TITLE="[Mono-bugs] [Bug 549926] Problem with exception handling with async (Web) Service calls">bugzilla_noreply at novell.com
       </A><BR>
    <I>Mon Nov  2 16:22:14 EST 2009</I>
    <P><UL>
        <LI>Previous message: <A HREF="094530.html">[Mono-bugs] [Bug 541898] ContextMenuStrip.SourceControl is always	null.
</A></li>
        <LI>Next message: <A HREF="094532.html">[Mono-bugs] [Bug 547151] UnKnown class exception - causing problems on Mac OS X Leopard
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#94531">[ date ]</a>
              <a href="thread.html#94531">[ thread ]</a>
              <a href="subject.html#94531">[ subject ]</a>
              <a href="author.html#94531">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE><A HREF="http://bugzilla.novell.com/show_bug.cgi?id=549926">http://bugzilla.novell.com/show_bug.cgi?id=549926</A>

User <A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">gnorton at novell.com</A> added comment
<A HREF="http://bugzilla.novell.com/show_bug.cgi?id=549926#c2">http://bugzilla.novell.com/show_bug.cgi?id=549926#c2</A>


Geoff Norton &lt;<A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">gnorton at novell.com</A>&gt; changed:

           What    |Removed                     |Added
----------------------------------------------------------------------------
                 CC|                            |<A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">gnorton at novell.com</A>
         AssignedTo|<A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">mono-bugs at lists.ximian.com</A>  |<A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">atsushi at ximian.com</A>




--- Comment #2 from Geoff Norton &lt;<A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">gnorton at novell.com</A>&gt;  2009-11-02 14:22:08 MST ---
Eno,

  Looking thru the code here, the issue is this, calling into a WCF service
with a callback set, but the service returns lets say a 403 or a 404, we fall
into this block:

                void GotResponse (IAsyncResult result)
                {
                        HttpChannelRequestAsyncResult channelResult =
(HttpChannelRequestAsyncResult) result.AsyncState;
                        channelResult.CompletedSynchronously &amp;=
result.CompletedSynchronously;

                        WebResponse res;
                        Stream resstr;
                        try {
                                res = web_request.EndGetResponse (result);
                                resstr = res.GetResponseStream ();
                        } catch (WebException we) {
                                res = we.Response;
#if NET_2_1 // debug
                                Console.WriteLine (we);
#endif
                                try {
                                        // The response might contain SOAP
fault. It might not.
                                        resstr = res.GetResponseStream ();
                                } catch (WebException we2) {
#if NET_2_1 // debug
                                        Console.WriteLine (we2);
#endif

Console.WriteLine (&quot;completing with we2&quot;);
                                        channelResult.Complete (we2);
                                        return;
                                }
                        }


When we call channelResult.Complete (we2), callback isn't set, so it never
bubbles up to user code.

-&gt; eno

-- 
Configure bugmail: <A HREF="http://bugzilla.novell.com/userprefs.cgi?tab=email">http://bugzilla.novell.com/userprefs.cgi?tab=email</A>
------- You are receiving this mail because: -------
You are the QA contact for the bug.
You are the assignee for the bug.
</PRE>





















<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="094530.html">[Mono-bugs] [Bug 541898] ContextMenuStrip.SourceControl is always	null.
</A></li>
	<LI>Next message: <A HREF="094532.html">[Mono-bugs] [Bug 547151] UnKnown class exception - causing problems on Mac OS X Leopard
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#94531">[ date ]</a>
              <a href="thread.html#94531">[ thread ]</a>
              <a href="subject.html#94531">[ subject ]</a>
              <a href="author.html#94531">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://lists.ximian.com/mailman/listinfo/mono-bugs">More information about the mono-bugs
mailing list</a><br>
</body></html>
